contracts/eosio.system/src/guilds.cpp

```diff
@@ -0,0 +1,67 @@
+#include <eosio.system/eosio.system.hpp>
+#include <eosio.token/eosio.token.hpp>
+#include <eosio.system/guilds.hpp>
+
+namespace eosiosystem {
+
+  using eosio::current_time_point;
+  using eosio::microseconds;
+  using eosio::token;
+
+  void system_contract::fundguilds() {
+    check(has_auth(guilds_account) || has_auth(_self), "Lacking OIG authority" );
+    const auto ct = current_time_point();
+    check(ct > guild_bp_pay_start_time, "Only available after guilds pay time");
+
+    fill_buckets();
+
+    guild_contract_state_singleton guild_contract_state(guilds_account, guilds_account.value);
+    auto guild_contract_status = guild_contract_state.get();
+
+    check(guild_contract_status.pending.amount <= max_daily_guilds_fund, "guild fund greater than daily limit");
+
+    guild_funding_state_singleton guild_funding_state(_self, _self.value);
+    auto guild_funding_status = guild_funding_state.get_or_create(_self);
+    check( ct - guild_funding_status.lst_fund > microseconds(useconds_per_day), "already funded within past day" );
```

Comment:
The guilds would like this claim action to come available at the same time everyday, so soemthing like `check( ct - guild_funding_status.lst_fund >= microseconds(useconds_per_day), "already funded within past day" );` should be better here to suit that requirement